superset/migrations/versions/4ce8df208545_migrate_time_range_for_default_filters.py

```diff
@@ -0,0 +1,94 @@
+"""empty message
+
+Revision ID: 4ce8df208545
+Revises: 55e910a74826
+Create Date: 2018-11-12 13:31:07.578090
+
+"""
+
+# revision identifiers, used by Alembic.
+import json
+
+from alembic import op
+from sqlalchemy import (
+    Column,
+    Integer,
+    Text,
+)
+from sqlalchemy.ext.declarative import declarative_base
+
+from superset import db
+
+revision = '4ce8df208545'
+down_revision = '55e910a74826'
+
+Base = declarative_base()
+
+
+class Dashboard(Base):
+    """Declarative class to do query in upgrade"""
+    __tablename__ = 'dashboards'
+    id = Column(Integer, primary_key=True)
+    json_metadata = Column(Text)
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = db.Session(bind=bind)
+
+    dashboards = session.query(Dashboard).all()
+    for i, dashboard in enumerate(dashboards):
+        print('scanning dashboard ({}/{}) >>>>'.format(i + 1, len(dashboards)))
+        if dashboard.json_metadata:
+            json_metadata = json.loads(dashboard.json_metadata)
+            has_update = False
+
+            # default_filters: it may define time_range filter
+            default_filters = json_metadata.get('default_filters')
+            if default_filters and default_filters != '{}':
+                try:
+                    filters = json.loads(default_filters)
+                    keys = [key for key, val in filters.items() if
+                            val.get('__from') or val.get('__to')]
+                    if len(keys):
+                        for key in keys:
+                            val = filters[key]
+                            val['__time_range'] = '{} : {}'.format(
```

Comment:
Is there any concern that `__time_range` could already exist? If so should it trump the `__from` and `__to`?

Reply:
yes, i think should check if `__time_range exist`. if it is exist, just abandon __from and __to parameter, so that not override existed __time_range.